src/python/vehicle.py

```diff
@@ -0,0 +1,183 @@
+from typing import Dict
+import tensorflow as tf
+import numpy as np
+import random
+import json
+
+
+def generate_loss_function(pathToSpec:str, functionName:str, resources:Dict[str,any]):
+    '''
+    pathToSpec: path to the vehicle spec .vcl file
+    functionName: name of the function for which we want to create the loss function
+    resources: dictionary mapping from the name of the resources declared in the spec to the python implementation
+    '''
+
+    jsonDICT = generateJson(pathToSpec, functionName)
+    context = []
+    loss = '_'
+    loss = translateJSONtoLOSSfunction(resources, jsonDICT)
+
+    return loss(context)
+    
+
+def generateJson(pathToSpec, functionName):
```

Comment:
I have snake_cased the remaining method, I didn't touch the variable names. Is it fine or do you want them too?
I have type annotated the arguments, what do you mean by the return types?
Something like `return jsonDICT:dict`?

Reply:
Yes, see the PEP 8 style guide (https://peps.python.org/pep-0008/#function-and-variable-names)

Reply:
> You put the return types after the colon in the signature, e.g. `def generateJson(pathToSpec, functionName): Dict[String, String]`

Looking online it should be more like `def generateJson(pathToSpec, functionName) -> Dict[String, String]:`
Also, for `generate_loss_function` and `to_loss_function` I have found the type `Callable` because there does not exist a `function` type.

Reply:
Yup, sounds good.